src/runtime/thread_pool.cc

```diff
@@ -297,6 +303,13 @@ class ThreadPool {
     return dmlc::ThreadLocalStore<ThreadPool>::Get();
   }
 
+  // order of CPU ids (most preferred first)
+  static std::vector<unsigned int> affinity_order;
```

Comment:
Ok, I think this choice was because I wasn't sure if it was safe to change affinity after threads were already created and the affinity has been set once. The original choice was to set everything up before the ThreadPool is constructed; as ThreadPool::ThreadLocal()->Configure seems to construct a ThreadPool if one does not already exist. Is it safe to mutate the threads_ in the backend impl after it has been created already?